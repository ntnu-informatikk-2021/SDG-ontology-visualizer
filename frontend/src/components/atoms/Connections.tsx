import { Button, Text, Wrap } from '@chakra-ui/react';
import React from 'react';
import { mapCorrelationToColor } from '../../common/node';
import { Node } from '../../types/ontologyTypes';

type ConnectionsProps = {
  connections: Array<Node>;
  titles: Array<string>;
  predicate: Array<string>;
  color: string;
  handleOnClick: (selectedConnection: Node, predicate: Array<string>) => void;
};
const Connections: React.FC<ConnectionsProps> = ({
  connections,
  titles,
  color,
  predicate,
  handleOnClick,
}: ConnectionsProps) => (
  <>
    <Text as="b" fontSize="lg">
      {connections.length ? titles[0] : titles[1]}
    </Text>
    <Wrap>
      {connections
        .sort((a, b) => b.correlation - a.correlation)
        .map((connection) => (
          <Button
            colorScheme="whiteAlpha"
            size="sm"
            bg={color + mapCorrelationToColor(connection.correlation)}
            key={connection.id}
            onClick={() => handleOnClick(connection, predicate)}
          >
            {connection.name}
          </Button>
        ))}
    </Wrap>
  </>
);

export default Connections;
