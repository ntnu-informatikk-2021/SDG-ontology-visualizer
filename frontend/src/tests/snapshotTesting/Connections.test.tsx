import React from 'react';
import renderer from 'react-test-renderer';
import Connections from '../../components/atoms/Connections';

const node2 = {
  id: 't2',
  name: 'hei',
  type: 'ok1',
  prefix: {
    prefix: 'jajaa',
    iri: '1234',
  },
  correlation: 3,
};

// const nodes = [node1, node2];

it('renders when there are no items', () => {
  const tree = renderer
    .create(
      <>
        <Connections
          connections={[node2]}
          titles={['Har positiv virkning til:', 'Har ingen etablerte positive pÃ¥virkninger enda']}
          color="green"
          handleOnClick={() => {
            console.log(node2);
          }}
        />
      </>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
