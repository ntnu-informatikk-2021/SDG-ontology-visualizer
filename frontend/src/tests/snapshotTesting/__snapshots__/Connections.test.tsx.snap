// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no items 1`] = `
Array [
  <b
    className="chakra-text css-6dvxm2"
  >
    Har positiv virkning til:
  </b>,
  <div
    className="chakra-wrap css-0"
  >
    <ul
      className="chakra-wrap__list css-umyqv7"
    >
      <button
        className="chakra-button css-19h3dke"
        onClick={[Function]}
        style={
          Object {
            "margin": 5,
          }
        }
        type="button"
      >
        hei
      </button>
    </ul>
  </div>,
]
`;
